### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by abhin2003**

### CODE REVIEW

File: src/Blog_App_frontend/src/AddBlog.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { useNavigate } from 'react-router-dom'; // Import useNavigate hook
+import './AddBlog.css'; // Import custom CSS file for styling
+
+// Define the validation schema using Yup
+const AddBlogSchema = Yup.object().shape({
+  title: Yup.string()
+    .max(125, 'Title must not exceed 125 characters')
+    .required('Title is required'),
+  description: Yup.string().required('Description is required'),
+});
```
Comment (mohitrajane) [inline src/Blog_App_frontend/src/AddBlog.jsx]: move to a constants file

Comment (mohitrajane) [reply src/Blog_App_frontend/src/AddBlog.jsx]: Change `AddBlogSchema` to `addBlogSchema`

File: src/Blog_App_frontend/src/AddBlog.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+function AddBlog({ handleAddBlog }) {
+  const navigate = useNavigate(); // Initialize useNavigate hook
+
+  return (
+    <div className="add-blog-container">
+      <h2 className="page-title">Add Blog</h2>
+      <Formik
+        initialValues={{ title: '', description: '' }}
```
Comment (mohitrajane) [inline src/Blog_App_frontend/src/AddBlog.jsx]: move initital vlaues to const file

File: src/Blog_App_frontend/src/AddBlog.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        validationSchema={AddBlogSchema}
+        onSubmit={async (values, { setSubmitting, resetForm }) => {
+          await handleAddBlog(values); // Wait for handleAddBlog to complete
+          setSubmitting(false);
+          resetForm();
+          navigate('/blogs'); // Redirect to Blogs page
```
Comment (mohitrajane) [inline src/Blog_App_frontend/src/AddBlog.jsx]: wrap in try catch,

File: src/Blog_App_frontend/src/AddBlog.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        }}
+      >
+        {({ isSubmitting }) => (
+          <Form className="add-blog-form">
+            <div className="form-group">
+              <label htmlFor="title">Title:</label>
+              <Field type="text" name="title" className="input" />
+              <ErrorMessage name="title" component="div" className="error-message" />
+            </div>
```
Comment (mohitrajane) [inline src/Blog_App_frontend/src/AddBlog.jsx]: use `<Form.Control.Feedback type="invalid">` from bootstrap instead of `ErrorMessage`

**Review by mohitrajane — CHANGES_REQUESTED** [review]

**Review by mohitrajane — APPROVED** [review]

File: src/Blog_App_frontend/src/AddBlog.jsx (hunk 1 above)
Comment (abhin2003) [reply src/Blog_App_frontend/src/AddBlog.jsx]: changed to addBlogs and moved the constants to a file constant.jsx

File: src/Blog_App_frontend/src/AddBlog.jsx (hunk 3 above)
Comment (abhin2003) [reply src/Blog_App_frontend/src/AddBlog.jsx]: ya done

File: src/Blog_App_frontend/src/AddBlog.jsx [hunk 5]
```diff
@@ -1,46 +1,63 @@
...
+import { initialValues } from './constants'; 
 
 // Define the validation schema using Yup
-const AddBlogSchema = Yup.object().shape({
+const addBlogSchema = Yup.object().shape({
   title: Yup.string()
     .max(125, 'Title must not exceed 125 characters')
     .required('Title is required'),
   description: Yup.string().required('Description is required'),
 });
```
Comment (mohitrajane) [inline src/Blog_App_frontend/src/AddBlog.jsx:17]: move this also to a constants file

**Review by mohitrajane — CHANGES_REQUESTED** [review]

File: src/Blog_App_frontend/src/AddBlog.jsx (hunk 2 above)
Comment (abhin2003) [reply src/Blog_App_frontend/src/AddBlog.jsx]: ok moved to constants.jsx

File: src/Blog_App_frontend/src/AddBlog.jsx (hunk 4 above)
Comment (abhin2003) [reply src/Blog_App_frontend/src/AddBlog.jsx]: changed it

File: src/Blog_App_frontend/src/AddBlog.jsx (hunk 5 above)
Comment (abhin2003) [reply src/Blog_App_frontend/src/AddBlog.jsx:17]: moved it to constants.jsx

**Pull request merged**